src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -2680,12 +2707,18 @@ static void top_toolbar_tool_update_scenery(int16_t x, int16_t y)
     {
         case SCENERY_TYPE_SMALL:
             gMapSelectFlags |= MAP_SELECT_FLAG_ENABLE;
-            if (gWindowSceneryClusterEnabled)
-            {
-                gMapSelectPositionA.x = mapTile.x - (8 << 5);
-                gMapSelectPositionA.y = mapTile.y - (8 << 5);
-                gMapSelectPositionB.x = mapTile.x + (7 << 5);
-                gMapSelectPositionB.y = mapTile.y + (7 << 5);
+            if (gWindowSceneryScatterEnabled)
+            {
+                uint16_t cluster_size = (gWindowSceneryScatterSize - 1) << 5;
+                gMapSelectPositionA.x = mapTile.x - cluster_size / 2;
+                gMapSelectPositionA.y = mapTile.y - cluster_size / 2;
+                gMapSelectPositionB.x = mapTile.x + cluster_size / 2;
+                gMapSelectPositionB.y = mapTile.y + cluster_size / 2;
+                if (gWindowSceneryScatterSize % 2 == 0)
+                {
+                    gMapSelectPositionB.x += (1 << 5);
```

Comment:
What's the significance of the `<< 5`?

Reply:
These shifts are already present in the code as seen in develop. I don't mind changing to `COORDS_XY_STEP`, but changing the data types feels a bit out of scope for this PR.